### ISSUE TITLE
Multiple use cases having common extensions

### ISSUE THREAD

**Issue by tayruxin** [nus-cs2103-AY2324S1/forum#415]
In our use cases across multiple commands, we have repeated extensions such as user keying in an invalid command or user keying in an invalid input. 

Will that be okay? Or what are some ways we can address this repetitiveness?

**Comment by damithc** [comment]
@tayruxin You can consider omitting those extensions and simply adding a note `Extensions X, Y and Z are handled similar to Un`

**Comment by tayruxin** [comment]
> @tayruxin You can consider omitting those extensions and simply adding a note `Extensions X, Y and Z are handled similar to Un`

Does this mean that we should create an extension scenario with the precondition being "the index is invalid" and the postcondition as "use case resumes at the specified steps"? as seen below: 

```
E1: Handle Invalid Input
- Preconditions: User input is invalid
- Main Flow:
  1. System displays an error message indicating the input is invalid.
  2. System prompts the user to correct the input.
- Postconditions: User receives feedback about the invalid input.
```
We will then reference this extension by E1. 

or..... should we have a separate use case for example, "Warn on invalid command" with the system being the actor

We would like to clarify which way of handling is better since duplicated exceptions are not mentioned explicitly in tutorials or the textbook. 

Thank you!

**Comment by damithc** [comment]
@tayruxin No need to extract out extensions separately. Specify it normally in one use case, and refer to that from other use cases.

For example, consider this use case from the textbook.

![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/1673303/888b4fe2-3940-4bd9-95fa-0ff37ab2e3b6)

In another use case, you could do the following:
```
...
...
Exensions:
    ...
    *a. At any time, User chooses to cancel the transfer -> handled similar to a* of UC23
```
While it is not a strict requirement, if you feel like you are needlessly repeating the some content, you can use something like the above (or some other sensible mechanism) to reduce such duplication.

**Comment by tayruxin** [comment]
Thank you!

**Comment by tayruxin** [comment]
>  *a. At any time, User chooses to cancel the transfer -> handled similar to a* of UC23

@damithc May I clarify if UC23 needs to be underlined?

**Comment by damithc** [comment]
> I clarify if UC23 needs to be underlined?

@tayruxin No. Underlining is only for use case inclusion. This is just referencing another use case.
